src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,19 @@
+package duke.task;
+
+import duke.task.Task;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.taskType = 'D';
```

Comment:
Avoid magic strings like 'D' - extract them out as constants and name them using proper naming conventions